packages/iris-grid/src/sidebar/visibility-ordering-builder/sortable-tree/SortableTree.tsx

```diff
@@ -0,0 +1,350 @@
+  const [overId, setOverId] = useState<UniqueIdentifier | null>(null);
+  const [offsetLeft, setOffsetLeft] = useState(0);
+
+  useEffect(() => {
+    setItems(defaultItems);
+  }, [defaultItems]);
+
+  const flattenedItems = useMemo(() => {
+    const flattenedTree = flattenTree(items);
+    const collapsedItems = flattenedTree.reduce<UniqueIdentifier[]>(
+      (acc, { children, collapsed, id }) =>
+        collapsed && children.length ? [...acc, id] : acc,
+      []
+    );
+
+    return removeChildrenOf(
+      flattenedTree,
+      activeId != null ? [activeId, ...collapsedItems] : collapsedItems
+    );
+  }, [activeId, items]);
+  const projected =
+    activeId != null && overId != null
+      ? getProjection(
+          flattenedItems,
+          activeId,
+          overId,
+          offsetLeft,
+          indentationWidth
+        )
+      : null;
+  const sensorContext: SensorContext = useRef({
+    items: flattenedItems,
+    offset: offsetLeft,
+  });
+  const [coordinateGetter] = useState(() =>
+    sortableTreeKeyboardCoordinates(sensorContext, indicator, indentationWidth)
+  );
+  const sensors = useSensors(
+    useSensor(PointerWithInteraction, {
+      activationConstraint: {
+        distance: 5,
+      },
+    })
+    // useSensor(KeyboardSensor, {
+    //   coordinateGetter,
+    // })
+  );
+
+  const sortedIds = useMemo(() => flattenedItems.map(({ id }) => id), [
+    flattenedItems,
+  ]);
+  const activeItem =
+    activeId != null ? flattenedItems.find(({ id }) => id === activeId) : null;
+
+  useEffect(() => {
+    sensorContext.current = {
+      items: flattenedItems,
+      offset: offsetLeft,
+    };
+  }, [flattenedItems, offsetLeft]);
+
+  return (
+    <DndContext
+      sensors={sensors}
+      collisionDetection={closestCenter}
+      measuring={measuring}
+      onDragStart={handleDragStart}
+      onDragMove={handleDragMove}
+      onDragOver={handleDragOver}
+      onDragEnd={handleDragEnd}
+      onDragCancel={handleDragCancel}
+    >
+      <SortableContext items={sortedIds} strategy={verticalListSortingStrategy}>
+        <div className={classNames(activeId != null && styles.MarchingAnts)}>
+          {flattenedItems.map(item => {
+            const { id, children, collapsed, depth } = item;
+            return (
+              <SortableTreeItem
+                key={id}
+                id={id}
+                value={id}
+                depth={id === activeId && projected ? projected.depth : depth}
+                indentationWidth={indentationWidth}
+                indicator={indicator}
+                collapsed={Boolean(collapsed && children.length)}
+                onCollapse={
+                  collapsible && children.length
+                    ? () => handleCollapse(id)
+                    : undefined
+                }
+                onRemove={
+                  removable && onRemove ? () => handleRemove(id) : undefined
```

Comment:
`handleRemove` undefined - I think you mean `onRemove(item)`.

Could also set `onRemove` to a default function (defined outside as a const) and then this could just be `removable ? onRemove(item) : undefined`

Reply:
I'm removing the removable and collapsible code from this. It was part of the example and we aren't using it. Should be easy enough to add back later if desired